libs/dsb-client-gateway-secrets-engine/src/lib/service/service/vault.service.ts

```diff
@@ -30,6 +31,52 @@ export class VaultService extends SecretsEngineService implements OnModuleInit {
     );
   }
 
+  public async getAllUsers(): Promise<UsersList> {
+    const res = await this.client.list(`${this.prefix}/${PATHS.USERS}`);
+
+    const keys: string[] = res.data.keys;
+
+    const usersToReturn: UsersList = [];
+
+    for (const key of keys) {
+      usersToReturn.push({
+        username: key,
+        password: await this.getUserAuthDetails(key),
+      });
+    }
+
+    return usersToReturn;
+  }
+
+  @Span('vault_getUserAuthDetails')
+  public async getUserAuthDetails(username: string): Promise<string> {
+    return this.client
+      .read(`${this.prefix}${PATHS.USERS}/${username}`)
+      .then(({ data }) => data.password)
+      .catch((err) => {
+        this.logger.error(err.message);
+
+        this.logger.error(err);
+
+        return null;
+      });
+  }
+
+  @Span('vault_setUserAuthDetails')
```

Comment:
What is this for? I know this is to set username and password to the vault but what is this for?

Reply:
Opentracing